src/tf/tf.cpp

```diff
@@ -336,6 +338,32 @@ struct tf_parser
         return prog.add_instruction(op, {args[0], weights});
     }
 
+    instruction_ref
+    parse_matmul(const std::string&, attribute_map attributes, std::vector<instruction_ref> args)
+    {
+        bool transa = false;
+        bool transb = false;
+
+        if(contains(attributes, "transpose_a"))
+        {
+            transa = attributes.at("transpose_a").b();
+        }
+        if(contains(attributes, "transpose_b"))
+        {
+            transb = attributes.at("transpose_a").b();
+        }
+
+        std::vector<int64_t> perm(args[0]->get_shape().lens().size());
+        std::iota(perm.begin(), perm.end(), int64_t{0});
+        // swap the last two elements
+        std::swap(*perm.rbegin(), *(perm.rbegin() + 1));
```

Comment:
Why do you need to swap the last two elements? Is that because it is column major?

Also, I think it would be cleaner to use `iter_swap`: `std::iter_swap(perm.end() - 1, perm.end() - 2)`.

Reply:
This was copied from our parse_gemm in onnx.

Reply:
In parse_gemm, we consider the case that input could be batch of matrices, and the last two dims are for the row and column of input matrices. That is the reason we swap the last two dims. As for MatMul, the ONNX operator does not support transposed matrix. So you need to check the TF spec to see whether input matrices can be transposed.

Reply:
From further investigation, tf.matmul is similar to numpy's matmul, so if the input is Nd, then the last two dimensions are the dimensions for the matrices. tf.matmul also suports transpose.